tests/test_formats.py

```diff
@@ -1060,6 +1062,511 @@ def test_format_number_with_sep_dec_marks():
+                "746,519.9 YiB",
+            ],
+        ),
+        (
+            "binary",
+            2,
+            True,
+            False,
+            True,
+            [
+                "−342 B",
+                "−3 B",
+                "0 B",
+                "0 B",
+                "0 B",
+                "1 B",
+                "500 B",
+                "512 B",
+                "994 B",
+                "1,000 B",
+                "1 KiB",
+                "2.19 GiB",
+                "769.12 TiB",
+                "225.18 PiB",
+                "59.98 YiB",
+                "746,519.93 YiB",
+            ],
+        ),
+        (
+            "binary",
+            3,
+            True,
+            False,
+            True,
+            [
+                "−342 B",
+                "−3 B",
+                "0 B",
+                "0 B",
+                "0 B",
+                "1 B",
+                "500 B",
+                "512 B",
+                "994 B",
+                "1,000 B",
+                "1 KiB",
+                "2.185 GiB",
+                "769.118 TiB",
+                "225.18 PiB",
+                "59.981 YiB",
+                "746,519.928 YiB",
+            ],
+        ),
+        (
+            "binary",
+            1,
+            False,
+            True,
+            False,
+            [
+                "−342B",
+                "−3B",
+                "0B",
+                "0B",
+                "0B",
+                "+1B",
+                "+500B",
+                "+512B",
+                "+994B",
+                "+1000B",
+                "+1KiB",
+                "+2.2GiB",
+                "+769.1TiB",
+                "+225.2PiB",
+                "+60YiB",
+                "+746519.9YiB",
+            ],
+        ),
+    ],
+)
+def test_fmt_bytes(
+    standard: str, decimals: int, use_seps: bool, force_sign: bool, incl_space: bool, x_out: str
+):
+    df = pd.DataFrame(
+        {
+            "x": [
+                -342.8,
+                -3,
+                0,
+                0.4,
+                0.9,
+                1,
+                500,
+                512,
+                994,
+                1000,
+                1024,
+                2346345274.3,
+                845653745232536,
+                253529876942760953,
+                72512895702785787335434345,
+                902487216348759693489128343269,
+            ]
+        }
+    )
+
+    gt = GT(df).fmt_bytes(
+        columns="x",
+        standard=standard,
+        decimals=decimals,
+        use_seps=use_seps,
+        force_sign=force_sign,
+        incl_space=incl_space,
+    )
+    x = _get_column_of_values(gt, column_name="x", context="html")
+    assert x == x_out
+
+
+# ------------------------------------------------------------------------------
+# Test `fmt_roman()`
+# ------------------------------------------------------------------------------
+
+
+@pytest.mark.parametrize(
```

Comment:
I strongly suspect you don't need to test that upper works on many variations of inputs values. (one is probably fine).